stravalib/model.py

```diff
@@ -81,40 +116,60 @@ def __repr__(self):
             attrs.append('id={0}'.format(self.id))
         if hasattr(self.__class__, 'name'):
             attrs.append('name={0!r}'.format(self.name))
-        if hasattr(self.__class__, 'resource_state') and self.resource_state is not None:
+        if (
+            hasattr(self.__class__, 'resource_state')
+            and self.resource_state is not None
+        ):
             attrs.append('resource_state={0}'.format(self.resource_state))
 
-        return '<{0}{1}>'.format(self.__class__.__name__, ' ' + ' '.join(attrs) if attrs else '')
+        return '<{0}{1}>'.format(
+            self.__class__.__name__, ' ' + ' '.join(attrs) if attrs else ''
+        )
 
 
 class ResourceStateEntity(BaseEntity):
     """
     Mixin for entities that include the resource_state attribute.
     """
-    resource_state = Attribute(int, (META, SUMMARY, DETAILED))  #: The detail-level for this entity.
+
+    resource_state = Attribute(
+        int, (META, SUMMARY, DETAILED)
+    )  #: The detail-level for this entity.
 
 
 class IdentifiableEntity(ResourceStateEntity):
     """
     Mixin for entities that include an ID attribute.
     """
-    id = Attribute(int, (META, SUMMARY, DETAILED))  #: The numeric ID for this entity.
+
+    id = Attribute(
```

Comment:
I am really confused here (maybe i am just confused but..) here we use ATTRIBUTE to create a class within another class identifiableEntity and then id= is used.

But [id is also a python function](https://docs.python.org/3/library/functions.html#id) 

so i'm just really getting confused by the nesting of elements in this module.

wondering if @hozn could help me understand. i know that there is some complexity in this package that I need to better understand but also i wonder if some of it could be simpler which might make maintenance easier? 

many thanks for any input @jsamoocha @hozn

Reply:
As I mentioned above, switching to Pydantic will remove much of this complexity.

Reply:
perfect.